shell/plugins/dashboard-store/mutations.js

```diff
@@ -86,8 +86,16 @@ export function createLoadArgs(ctx, dataType) {
   };
 }
 
+/**
+ * Add or update the given resource in the store
+ *
+ * invalidatePageCache
+ * - if something calls `load` then the cache no longer has a page so we invalidate it
+ * - however on resource create or remove this can lead to lists showing nothing... before the new page is fetched
```

Comment:
As in the page appears to already be loaded with an empty list?

It really seems like we should just be requesting state on the page level and tunneling it down the props instead of adding more stateful components to our app.

Reply:
Correct, ish.

Create - The user returns from create page to an empty list which is then populated
Remove - All rows in the list are removed, which are then populated again. This also allows us to show pre-delete state like pod 'terminating'

> it really seems like we should just be requesting state on the page level and tunneling it down the props instead of adding more stateful components to our app.

Note quite sure i follow. Could you expand?